compiler/modules/bir/bir.bal

```diff
@@ -499,7 +569,7 @@ public type CallInsn readonly & record {|
 public type AssignInsn readonly & record {|
     *InsnBase;
     INSN_ASSIGN name = INSN_ASSIGN;
-    Register result;
+    TempRegister|VarRegister|FinalRegister result;
```

Comment:
Where do we use AssignInsn with a TempRegister? That's not great because we want temp registers to be single-assignment,

Reply:
Currently we assign to tempRegisters "internally" in the fallowing places

1. https://github.com/ballerina-platform/nballerina/blob/0005efda65f580233f63a8485cd5ad774e5feb4d/compiler/modules/front/stmt.bal#L1161
2. https://github.com/ballerina-platform/nballerina/blob/0005efda65f580233f63a8485cd5ad774e5feb4d/compiler/modules/front/expr.bal#L312
3. https://github.com/ballerina-platform/nballerina/blob/0005efda65f580233f63a8485cd5ad774e5feb4d/compiler/modules/front/expr.bal#L317
4. https://github.com/ballerina-platform/nballerina/blob/0005efda65f580233f63a8485cd5ad774e5feb4d/compiler/modules/front/expr.bal#L696
5. https://github.com/ballerina-platform/nballerina/blob/0005efda65f580233f63a8485cd5ad774e5feb4d/compiler/modules/front/expr.bal#L698

Reply:
1 is fine (it's just initialization).  The others need fixing (in separate PR).

Reply:
Actually they are OK, they are doing the assignments in alternative branches.